lib/screens/auth/email_sign_in_screen.dart

```diff
@@ -7,18 +7,25 @@ import 'package:provider/provider.dart';
 import 'package:planner/services/user_auth_service.dart';
 import 'package:planner/widgets/policy.dart';
 
-enum Status { Initial, New, Registered}
+enum Status { Initial, New, Registered }
 
 const MIN_PASSWORD_LENGTH = 6;
 
 final emailRegex = new RegExp(
     r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"); // https://emailregex.com/
 
 class EmailSignInScreen extends StatelessWidget {
+  final Function onBackClick;
+
+  EmailSignInScreen({required this.onBackClick});
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(title: Text('Email')),
+      appBar: AppBar(
+          title: Text('Email'),
+          leading: IconButton(
+              icon: Icon(Icons.arrow_back), onPressed: () => onBackClick())),
```

Comment:
Can just do `onPressed: onBackClick`

Reply:
```suggestion
              onPressed: onBackClick())),
```
works, if thats what you mean, done

Reply:
It's because the `onBackClick` field on this widget has been declared too generic, so it doesn't know if `onBackClick` takes in arguments or not, so it can't guarantee that it's a no-argument function (what `onPressed` requires

However, trying to see what is the inferred type of that function in actual uses of them shows
![image](https://user-images.githubusercontent.com/25349581/128636585-457d5df3-9556-4768-94f2-e04a3ac7c97b.png)

`void Function()` as being the type of the function. To not make it too specific on what the return type of the function is, you can change the type of `onBackClick` in `email_sign_in_screen.dart` to `Function()` instead of `Function` and then it should work.

Reply:
good idea.